plugins/modules/elastic_endpoint_security.py

```diff
@@ -70,6 +64,52 @@ def create_securityctrl_baseline_settings(self,pkg_policy_object):
+          pkg_policy_object = self.toggle_pkg_policy_stream_onoff(pkg_policy_object, 'linux.pageinfo', False)
+          pkg_policy_info = self.update_pkg_policy(pkg_policy_object_id,pkg_policy_object)
+          
+        elif pkg_policy_object['package']['title'] == 'Windows':
+          pkg_policy_object_id = pkg_policy_object['id']
+          pkg_policy_object = self.toggle_pkg_policy_input_onoff(pkg_policy_object, 'winlog', False)
+          pkg_policy_object = self.toggle_pkg_policy_input_onoff(pkg_policy_object, 'httpjson', False)
+          pkg_policy_object = self.toggle_pkg_policy_input_onoff(pkg_policy_object, 'windows/metrics', True)
+          pkg_policy_object = self.toggle_pkg_policy_stream_onoff(pkg_policy_object, 'windows.perfmon', False)
+          pkg_policy_object = self.toggle_pkg_policy_stream_onoff(pkg_policy_object, 'windows.service', True)
+          pkg_policy_info = self.update_pkg_policy(pkg_policy_object_id,pkg_policy_object)
+          
+        elif pkg_policy_object['package']['title'] == 'System':
+          pkg_policy_object_id = pkg_policy_object['id']
+          pkg_policy_object = self.toggle_pkg_policy_input_onoff(pkg_policy_object, 'logfile', False)
+          pkg_policy_object = self.toggle_pkg_policy_input_onoff(pkg_policy_object, 'winlog', False)
+          pkg_policy_object = self.toggle_pkg_policy_input_onoff(pkg_policy_object, 'system/metrics', True)
+          pkg_policy_object = self.toggle_pkg_policy_input_onoff(pkg_policy_object, 'httpjson', False)
+          pkg_policy_object = self.toggle_pkg_policy_stream_onoff(pkg_policy_object, 'system.load', True)
+          pkg_policy_object = self.toggle_pkg_policy_stream_onoff(pkg_policy_object, 'system.core', False)
+          pkg_policy_object = self.toggle_pkg_policy_stream_onoff(pkg_policy_object, 'system.process.summary', True)
+          pkg_policy_object = self.toggle_pkg_policy_stream_onoff(pkg_policy_object, 'system.socket_summary', True)  
+          pkg_policy_object = self.toggle_pkg_policy_stream_onoff(pkg_policy_object, 'system.uptime', True)
+          pkg_policy_object = self.toggle_pkg_policy_stream_onoff(pkg_policy_object, 'system.network', True)  
+          pkg_policy_object = self.toggle_pkg_policy_stream_onoff(pkg_policy_object, 'system.cpu', True)
+          pkg_policy_object = self.toggle_pkg_policy_stream_onoff(pkg_policy_object, 'system.filesystem', True)  
+          pkg_policy_object = self.toggle_pkg_policy_stream_onoff(pkg_policy_object, 'system.fstat', True)
+          pkg_policy_object = self.toggle_pkg_policy_stream_onoff(pkg_policy_object, 'system.process', True)  
+          pkg_policy_object = self.toggle_pkg_policy_stream_onoff(pkg_policy_object, 'system.memory', True)
+          pkg_policy_object = self.toggle_pkg_policy_stream_onoff(pkg_policy_object, 'system.diskio', True)  
+          pkg_policy_info = self.update_pkg_policy(pkg_policy_object_id,pkg_policy_object)
+
```

Comment:
It looks like we're specifying actual policy settings within the module itself here.  That should be an input so that the module remains generic, with a settings dictionary passed in from the ansible playbook to determine which settings to apply.

Reply:
The entire configuration for Linux and System are quite large. I imagine it can get quite a bit larger for other Integrations. I have added the ability to add the JSON to the ansible job if you wish to pass the entire configuration instead of updating an element at a time. It does not appear to be possible to just send a variable or two to update via the REST API so using a couple functions to get the entire configuration, update specific variables, and then running a standard update or passing the entire configuration as an argument seems to be best options.